src/log.c

```diff
@@ -0,0 +1,92 @@
+#include <string.h>
+#include <stdarg.h>
+#include <syslog.h>
+#include <errno.h>
+
+#include "gophernicus.h"
+
+#if defined(LOG_UPTO)
+#define _LOG_UPTO LOG_UPTO
+#else
+#define _LOG_UPTO(pri)							\
+	(								\
+	((pri) == LOG_EMERG)   ? (LOG_MASK(LOG_EMERG)) :		\
+	((pri) == LOG_ALERT)   ? (LOG_MASK(LOG_EMERG) | LOG_MASK(LOG_ALERT)) : \
+	((pri) == LOG_CRIT)    ? (LOG_MASK(LOG_EMERG) | LOG_MASK(LOG_ALERT) | LOG_MASK(LOG_CRIT)) : \
+	((pri) == LOG_ERR)     ? (LOG_MASK(LOG_EMERG) | LOG_MASK(LOG_ALERT) | LOG_MASK(LOG_CRIT) | LOG_MASK(LOG_ERR)) : \
+	((pri) == LOG_WARNING) ? (LOG_MASK(LOG_EMERG) | LOG_MASK(LOG_ALERT) | LOG_MASK(LOG_CRIT) | LOG_MASK(LOG_ERR) | LOG_MASK(LOG_WARNING)) : \
+	((pri) == LOG_NOTICE)  ? (LOG_MASK(LOG_EMERG) | LOG_MASK(LOG_ALERT) | LOG_MASK(LOG_CRIT) | LOG_MASK(LOG_ERR) | LOG_MASK(LOG_WARNING) | LOG_MASK(LOG_NOTICE)) : \
+	((pri) == LOG_INFO)    ? (LOG_MASK(LOG_EMERG) | LOG_MASK(LOG_ALERT) | LOG_MASK(LOG_CRIT) | LOG_MASK(LOG_ERR) | LOG_MASK(LOG_WARNING) | LOG_MASK(LOG_NOTICE) | LOG_MASK(LOG_INFO)) : \
+	((pri) == LOG_DEBUG)   ? (LOG_MASK(LOG_EMERG) | LOG_MASK(LOG_ALERT) | LOG_MASK(LOG_CRIT) | LOG_MASK(LOG_ERR) | LOG_MASK(LOG_WARNING) | LOG_MASK(LOG_NOTICE) | LOG_MASK(LOG_INFO) | LOG_MASK(LOG_DEBUG)) : \
+	0)
+#endif
+
+static int _enable = 0;
```

Comment:
I don't really like this global, but I guess it's ok, since it's only changed once by log_init.